enhancements/microshift/etcd-as-a-transient-systemd-unit.md

```diff
@@ -0,0 +1,202 @@
+    
+The actual execution of etcd in the transient systemd unit will also be considered in this enhancement; there are currently two scenarios (in detail below): running the etcd container image under podman, and running the etcd binary installed via an RPM.
+    
+#### etcd Container Image with Podman
+This scenario would run the etcd server as a container executed via a `podman` command.
+    
+##### Podman Pros
+* This would be similar to how etcd is built and shipped for OCP and would reuse the existing build machinery for openshift/etcd.
+    
+##### Podman Cons
+* The current etcd image is too large, ~400MB, either it will need to be shrunk or a new image will need to be created with the bare minimum in it; the etcd binary alone is about 30MB.
+* This scenario places a new runtime dependency on podman; it would need to be installed and ready before MicroShift could be used.
+* MicroShift and its dependencies (incl. microshift-networking) are deliberately delivered as RPMs, so they natively fit into customers' build pipeline and content delivery for the RHEL4Edge rpm-ostrees they're building. Introducing non-RPM dependencies for MicroShift will complicate this.
+* A (small) risk of conflicts when customers use their own management agent for managing Podman workloads.
+    
+#### etcd Binary from an RPM
+This will lay the etcd binary down into the same directory as the MicroShift binary so that in a development environment a local build of etcd will be used.
+
+The RPM will be built in the same build root as MicroShift as a go submodule. This will resolve the dependency issues, but will cause etcd to be built with an unexpected version of Golang - Upstream/OpenShift etcd expects Go1.16, MicroShift is built with Go1.19.
+While this is not ideal, decoupling etcd into its own process to resolve the dependency conflict is the first step and getting the etcd Go build version aligned back to 1.16 can be a future goal.
+
+##### RPM Pros
+* This scenario should incur the least amount of non-etcd-related overhead since the binary is being run directly.
+* Integration is straight-forward; the etcd rpm is another MicroShift dependency, embedded into the rpm-ostree and distributed the same way as other dependencies.
+* Clear, consistent architecture: There is only either RPM-installed content or content hosted on the cluster - no third way of installing/running things.
+    
+##### RPM Cons
+* New plumbing will need to be created to (or potentially existing will need to be updated/refurbished to) build etcd and package it into an RPM.
+* etcd will need to be built with a newer version of Golang (1.19) than is currently expected from upstream (1.16); etcd will be built under the MicroShift build root, locking the Golang version.
+
+### Workflow Description
+
+TODO
+
+#### Variation [optional]
+
+TODO
+
+### API Extensions
+
+TODO
+
+### Implementation Details/Notes/Constraints [optional]
+
+TODO
+
+### Risks and Mitigations
+
+TODO
```

Comment:
One risk is that we could have an issue if the etcd process does not exit and we try to start it again. What are we doing to protect against that situation? Do we need to do something other than relying on systemd?